packages/functional-tests/tests/react-conversion/oauthResetPassword.spec.ts

```diff
@@ -2,285 +2,337 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { test, expect } from '../../lib/fixtures/standard';
+import { expect, test } from '../../lib/fixtures/standard';
 import { EmailHeader, EmailType } from '../../lib/email';
 import { syncMobileOAuthQueryParams } from '../../lib/query-params';
 import { LoginPage } from '../../pages/login';
-import { RelierPage } from '../../pages/relier';
-import { ResetPasswordPage } from '../../pages/resetPassword';
-import { BaseTarget, Credentials } from '../../lib/targets/base';
+import { ResetPasswordReactPage } from '../../pages/resetPasswordReact';
+import { BaseTarget } from '../../lib/targets/base';
 import { Page } from '@playwright/test';
 
+const SERVICE_NAME_123 = '123';
+const SERVICE_NAME_FIREFOX = 'Firefox';
+
 test.describe('severity-1 #smoke', () => {
   test.describe('oauth reset password react', () => {
-    let resetPasswordReactFlag = false;
-
-    test.beforeEach(async ({ pages: { configPage, resetPassword } }) => {
-      resetPasswordReactFlag = resetPassword.react;
-      resetPassword.react = true;
+    test.beforeEach(async ({ pages: { configPage } }) => {
       test.slow();
 
       const config = await configPage.getConfig();
       test.skip(config.showReactApp.resetPasswordRoutes !== true);
     });
 
-    test.afterEach(async ({ pages: { resetPassword } }) => {
-      resetPassword.react = resetPasswordReactFlag;
-    });
-
     test('reset password', async ({
       target,
       page,
       credentials,
-      pages: { login, relier, resetPassword, settings },
+      pages: { login, relier, resetPasswordReact },
     }) => {
-      await openRelier({ pages: { settings, relier } });
-      await passwordResetFlow(
-        {
-          target,
-          page,
-          credentials,
-          pages: { login, relier, resetPassword },
-        },
-        {}
+      // Make sure user is not signed in, and goes to the relier (ie 123done)
+      await relier.goto('showReactApp=true');
+
+      await relier.clickEmailFirst();
+
+      await beginPasswordReset(
+        page,
+        login,
+        resetPasswordReact,
+        credentials.email,
+        SERVICE_NAME_123
+      );
+
+      const link = await getConfirmationEmail(
+        target,
+        resetPasswordReact,
+        credentials.email
       );
+
+      await page.goto(link, { waitUntil: 'load' });
+      await resetPasswordReact.fillOutNewPasswordForm(credentials.password);
+
+      // Note: We used to redirect the user back to the relier in some cases
+      // but we've decided to just show the success message for now
+      // and let the user re-authenticate with the relier.
+      await page.waitForURL(/reset_password_verified/);
+      expect(resetPasswordReact.passwordResetConfirmationHeading).toBeVisible();
+      expect(
+        page.getByText(new RegExp(`.*${SERVICE_NAME_123}.*`, 'i'))
+      ).toBeVisible();
```

Comment:
🤦🏻‍♀️ This is a good catch, thank you very much.
There is a linting rule that can help with this playwright/missing-playwright-await. I've activated it and applied the fixes (see https://github.com/mozilla/fxa/commit/698210b7903c78d68e3fef89a7e08cda31fe5366). This will add scope to the PR, but since you are seeing failures locally, probably best to address this everywhere earlier.

Reply:
Niiiice! Thank you